src/SFA.DAS.EmployerApprenticeshipsService.Application/Commands/DeleteApprentice/DeleteApprenticeshipCommandHandler.cs

```diff
@@ -37,8 +34,20 @@ IValidator<DeleteApprenticeshipCommand> validator
                 throw new InvalidRequestException(validationResult.ValidationDictionary);
             }
 
-            await _commitmentsService.DeleteEmployerApprenticeship(message.AccountId, message.ApprenticeshipId,
+            var apprenticeship = await _commitmentsApi.GetEmployerApprenticeship(message.AccountId, message.ApprenticeshipId);
```

Comment:
The only consumer of this command (EmployerCommitmentsOrchestrator.DeleteApprenticeship) already retrieves the Apprenticeship, so that could be passed in the message, instead of just the ApprenticeshipId, in a similar manner to the UpdateApprenticeshipCommandHandler below. That would save fetching the Apprenticeship twice from the db.

However, it looks like EmployerCommitmentsOrchestrator.DeleteApprenticeship fetches the apprenticeship, but does nothing with it. It could possibly be fetching it as a check that it exists (maybe?), but it looks like it's doing it unnecessarily to me and the fetch could be removed!

I think we should only fetch it once though, whichever way we go.

Reply:
Removed unused apprenticeship fetch from consumer